contracts/cosmwasm-vm/cw-ibc-core/src/contract.rs

```diff
@@ -31,14 +30,500 @@ impl<'a> CwIbcCoreContext<'a> {
         info: MessageInfo,
         msg: ExecuteMsg,
     ) -> Result<Response, ContractError> {
-        todo!()
+        match msg {
+            ExecuteMsg::RegisterClient {
+                client_type,
+                client_address,
+            } => {
+                self.check_sender_is_owner(deps.as_ref().storage, info.sender.clone())
+                    .unwrap();
```

Comment:
looks like it already throws contract error so lets propagate via  "?;" to indicate that the error is compatible.